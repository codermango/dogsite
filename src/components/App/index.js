import React from 'react';
import Breed from '../Breed';
import DataProvider from '../DataProvider';
import styles from './styles.module.css';


const App = () => {
  return (
    <DataProvider>
      {({ data, isLoading, error}) => {

        if (error) {
          return <div>{error}</div>;
        }
        if (!data || isLoading) {
          return <div className={styles.loading}>Loading...</div>;
        }
        return (
          <div>
            {data.map((breed) => <Breed key={breed.name} data={breed} />)}
          </div>
        );
      }}
    </DataProvider>
  );
}

export default App;
