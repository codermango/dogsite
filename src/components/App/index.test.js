import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './index.js';


describe('App', () => {
  it('render loading correctly', () => {
    const state1 = { data: null, error: null, isLoading: true };
    const state2 = { data: null, error: null, isLoading: false };
    const component = shallow(<App />);
    const childrenComp1 = shallow(component.props().children(state1));
    const childrenComp2 = shallow(component.props().children(state2));

    expect(toJson(childrenComp1)).toMatchSnapshot();
    expect(toJson(childrenComp2)).toMatchSnapshot();
  });

  it('render error correctly', () => {
    const state = { data: null, error: 'error', isLoading: false }
    const component = shallow(<App />);
    const childrenComp = shallow(component.props().children(state));
    expect(toJson(childrenComp)).toMatchSnapshot();
  });

  it('render data correctly', () => {
    const state = { data: [{ name: 'testName1', image: 'testImage1' }], error: null, isLoading: false }
    const component = shallow(<App />);
    console.log(component.props().children(state))
    const childrenComp = shallow(component.props().children(state));
    expect(toJson(childrenComp)).toMatchSnapshot();
  });
});

