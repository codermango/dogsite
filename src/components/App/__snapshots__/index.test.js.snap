// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render data correctly 1`] = `
<div>
  <Breed
    data={
      Object {
        "image": "testImage1",
        "name": "testName1",
      }
    }
    key="testName1"
  />
</div>
`;

exports[`App render error correctly 1`] = `
<div>
  error
</div>
`;

exports[`App render loading correctly 1`] = `
<div
  className="loading"
>
  Loading...
</div>
`;

exports[`App render loading correctly 2`] = `
<div
  className="loading"
>
  Loading...
</div>
`;
